ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/sync/BlockPropagationManager.java

```diff
@@ -496,7 +496,8 @@ private CompletableFuture<Block> maybeRepeatGetBlock(
             .orElseGet(() -> blockchain.getBlockByNumber(blockNumber));
 
     // check if we got this block by other means
-    if (maybeBlock.isPresent()) {
+    if (maybeBlock.isPresent()
+        || (maybeBlockHash.isPresent() && pendingBlocksManager.contains(maybeBlockHash.get()))) {
```

Comment:
this seems unrelated to peering - is there tests around this?

Reply:
maybe we could put this fix in a different PR?